src/pybind/mgr/cephadm/templates/services/prometheus/prometheus.yml.j2

```diff
@@ -149,18 +147,31 @@ scrape_configs:
 {% endif %}
 {% endif %}
 
-{% if not secure_monitoring_stack %}
+{% for url, details in clusters_credentials.items() %}
   - job_name: 'federate'
     scrape_interval: 15s
     honor_labels: true
     metrics_path: '/federate'
+    relabel_configs:
+    - source_labels: [__address__]
+      target_label: cluster
+      replacement: {{ cluster_fsid }}
+{% if secure_monitoring_stack %}
+    scheme: https
+    tls_config:
+      ca_file: {{ details['cert_file_name'] }}
+    basic_auth:
+      username: {{ details['user'] }}
+      password: {{ details['password'] }}
+{% endif %}
     params:
       'match[]':
         - '{job="ceph"}'
         - '{job="node"}'
         - '{job="haproxy"}'
         - '{job="ceph-exporter"}'
+        - '{job="nvmeof"}'
```

Comment:
this line sounds unrelated to this change, right? was this a BUG in the previous version? if it's the case please create a separate PR for it :+1:

Reply:
This is actually intended, This section defines the criteria for selecting which job metrics to retrieve from the target cluster. Since we have the nvmeOf metrics available as well, so added that filter here as well.